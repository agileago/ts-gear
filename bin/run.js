"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
exports.__esModule = true;
var fs_1 = require("fs");
var path_1 = require("path");
var definitions_1 = require("./definitions");
var fetchSwagger_1 = require("./fetchSwagger");
var requests_1 = require("./requests");
var prettierWrite_1 = require("./prettierWrite");
var util_1 = require("./util");
var option_1 = require("./option");
var serviceIndexFileContent_1 = require("./serviceIndexFileContent");
var warningComment_1 = require("./warningComment");
var interceptorFilePath = path_1.resolve(util_1.tsGearRoot, 'src/interceptor.ts');
/**
 * step 1: get user config
 * step 2: fetch schema
 * step 3: parse schema to ts template content
 * step 4: write ts file
 * */
exports.run = function () { return __awaiter(void 0, void 0, void 0, function () {
    var cwd, userConfig, config, dest, projectNamesFromCommandLine, projects, _a, _b, _i, i, project, projectPath, schema, _c, $refsNotInDefinitions, $refsInPaths, $refsTypes, definitions, definitionsPath, _d, requestsContent, mockRequestsContent, requestFilePath, mockResponsePath, indexFilePath, projectInterceptorFile;
    return __generator(this, function (_e) {
        switch (_e.label) {
            case 0:
                cwd = process.cwd();
                userConfig = require(path_1.join(cwd, 'ts-gear'));
                config = (userConfig["default"] ? userConfig["default"] : userConfig);
                dest = path_1.join(cwd, config.dest);
                if (!fs_1.existsSync(dest)) {
                    fs_1.mkdirSync(dest);
                }
                projectNamesFromCommandLine = option_1.getProjectsFromCommmandLine();
                projects = config.projects;
                if (projectNamesFromCommandLine.length > 0) {
                    projects = projectNamesFromCommandLine
                        .map(function (name) {
                        return config.projects.find(function (p) { return p.name === name; });
                    })
                        .filter(Boolean);
                }
                _a = [];
                for (_b in projects)
                    _a.push(_b);
                _i = 0;
                _e.label = 1;
            case 1:
                if (!(_i < _a.length)) return [3 /*break*/, 12];
                i = _a[_i];
                if (!projects.hasOwnProperty(i)) {
                    return [3 /*break*/, 11];
                }
                project = projects[i];
                projectPath = path_1.join(dest, project.name);
                // 在dest文件夹内建立项目文件夹
                if (!fs_1.existsSync(projectPath)) {
                    fs_1.mkdirSync(projectPath);
                }
                return [4 /*yield*/, fetchSwagger_1.fetchSwaggerJSONSchema(project, project.fetchOption)];
            case 2:
                schema = _e.sent();
                return [4 /*yield*/, util_1.initializeSchema(schema)];
            case 3:
                _c = _e.sent(), $refsNotInDefinitions = _c.$refsNotInDefinitions, $refsInPaths = _c.$refsInPaths;
                return [4 /*yield*/, definitions_1.transform$RefsNotInDefinitions($refsNotInDefinitions)
                    // 生成definitions
                ];
            case 4:
                $refsTypes = _e.sent();
                return [4 /*yield*/, definitions_1.transformDefinitionsToTypescript(schema.definitions)];
            case 5:
                definitions = _e.sent();
                definitionsPath = path_1.join(projectPath, 'definitions.ts');
                return [4 /*yield*/, prettierWrite_1.prettierWrite(definitionsPath, warningComment_1.warningComment + definitions + $refsTypes)
                    // 生成request函数与mock request数据
                ];
            case 6:
                _e.sent();
                return [4 /*yield*/, requests_1.generateRequests(schema, $refsInPaths, project.pathMatcher)];
            case 7:
                _d = _e.sent(), requestsContent = _d.requestsContent, mockRequestsContent = _d.mockRequestsContent;
                requestFilePath = path_1.join(projectPath, 'request.ts');
                return [4 /*yield*/, prettierWrite_1.prettierWrite(requestFilePath, warningComment_1.warningComment + requestsContent)];
            case 8:
                _e.sent();
                mockResponsePath = path_1.join(projectPath, 'mockRequest.ts');
                return [4 /*yield*/, prettierWrite_1.prettierWrite(mockResponsePath, warningComment_1.warningComment + mockRequestsContent)];
            case 9:
                _e.sent();
                indexFilePath = path_1.join(projectPath, 'index.ts');
                return [4 /*yield*/, prettierWrite_1.prettierWrite(indexFilePath, warningComment_1.warningComment + serviceIndexFileContent_1.serviceIndexFileContent)
                    // 每个项目的拦截器文件只在第一次生成时copy一次
                    // 这个文件可能会写入一些请求的配置
                    // 不应该被覆盖
                ];
            case 10:
                _e.sent();
                projectInterceptorFile = path_1.join(projectPath, 'interceptor.ts');
                if (!fs_1.existsSync(projectInterceptorFile)) {
                    fs_1.copyFileSync(interceptorFilePath, projectInterceptorFile);
                }
                _e.label = 11;
            case 11:
                _i++;
                return [3 /*break*/, 1];
            case 12: return [2 /*return*/];
        }
    });
}); };
