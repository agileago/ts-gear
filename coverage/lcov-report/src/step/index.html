
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/step</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/step</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.24% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>195/290</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">49.17% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>59/120</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">62.07% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>36/58</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>191/285</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="assembleSchemaToGlobal.ts"><a href="assembleSchemaToGlobal.ts.html">assembleSchemaToGlobal.ts</a></td>
	<td data-value="96.55" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.55" class="pct high">96.55%</td>
	<td data-value="29" class="abs high">28/29</td>
	<td data-value="91.67" class="pct high">91.67%</td>
	<td data-value="12" class="abs high">11/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="96.43" class="pct high">96.43%</td>
	<td data-value="28" class="abs high">27/28</td>
	</tr>

<tr>
	<td class="file medium" data-value="cleanRefAndDefinitionName.ts"><a href="cleanRefAndDefinitionName.ts.html">cleanRefAndDefinitionName.ts</a></td>
	<td data-value="78.57" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 78%"></div><div class="cover-empty" style="width: 22%"></div></div>
	</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	</tr>

<tr>
	<td class="file high" data-value="colletRefsInRequestAndPatchDefinition.ts"><a href="colletRefsInRequestAndPatchDefinition.ts.html">colletRefsInRequestAndPatchDefinition.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

<tr>
	<td class="file high" data-value="fetchSwagger.ts"><a href="fetchSwagger.ts.html">fetchSwagger.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	</tr>

<tr>
	<td class="file low" data-value="generateDefinitionContent.ts"><a href="generateDefinitionContent.ts.html">generateDefinitionContent.ts</a></td>
	<td data-value="19.35" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 19%"></div><div class="cover-empty" style="width: 81%"></div></div>
	</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	</tr>

<tr>
	<td class="file low" data-value="importAllDefinition.ts"><a href="importAllDefinition.ts.html">importAllDefinition.ts</a></td>
	<td data-value="22.22" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 22%"></div><div class="cover-empty" style="width: 78%"></div></div>
	</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file high" data-value="parseGenericType.ts"><a href="parseGenericType.ts.html">parseGenericType.ts</a></td>
	<td data-value="98.63" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 98%"></div><div class="cover-empty" style="width: 2%"></div></div>
	</td>
	<td data-value="98.63" class="pct high">98.63%</td>
	<td data-value="73" class="abs high">72/73</td>
	<td data-value="86.84" class="pct high">86.84%</td>
	<td data-value="38" class="abs high">33/38</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="98.57" class="pct high">98.57%</td>
	<td data-value="70" class="abs high">69/70</td>
	</tr>

<tr>
	<td class="file medium" data-value="prepareProjectDirectory.ts"><a href="prepareProjectDirectory.ts.html">prepareProjectDirectory.ts</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

<tr>
	<td class="file low" data-value="translateSchema.ts"><a href="translateSchema.ts.html">translateSchema.ts</a></td>
	<td data-value="20" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="45" class="abs low">9/45</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="20.45" class="pct low">20.45%</td>
	<td data-value="44" class="abs low">9/44</td>
	</tr>

<tr>
	<td class="file low" data-value="writeProject.ts"><a href="writeProject.ts.html">writeProject.ts</a></td>
	<td data-value="35.71" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 35%"></div><div class="cover-empty" style="width: 65%"></div></div>
	</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Apr 27 2020 18:43:36 GMT+0800 (GMT+08:00)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    