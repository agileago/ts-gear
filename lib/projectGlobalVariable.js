"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const clearObject_1 = require("./tool/clearObject");
const projectGlobal = {};
exports.getGlobal = (project) => {
    if (!projectGlobal[project.name]) {
        projectGlobal[project.name] = {
            definitionMap: {},
            requestMap: {},
            requestRefSet: new Set(),
        };
    }
    return projectGlobal[project.name];
};
exports.restore = (project) => {
    const g = projectGlobal[project.name];
    clearObject_1.clearObject(g.definitionMap);
    clearObject_1.clearObject(g.requestMap);
    g.requestRefSet.clear();
};
exports.httpMethods = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch'];
