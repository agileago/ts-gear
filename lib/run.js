"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.run = exports.processProject = void 0;
const step = require("./step");
const projectGlobalVariable_1 = require("./projectGlobalVariable");
exports.processProject = async (project, tsGearConfigPath) => {
    step.prepareProjectDirectory(project, tsGearConfigPath);
    const spec = await step.fetchSwagger(project, tsGearConfigPath);
    if (project.translationEngine) {
        await step.translateSchema(spec, project.translationEngine);
    }
    const keepGeneric = project.keepGeneric !== false;
    step.cleanRefAndDefinitionName(spec, keepGeneric);
    step.assembleSchemaToGlobal(spec, project);
    if (keepGeneric) {
        step.parseGenericType(project);
    }
    step.collectRefsInRequestAndPatchDefinition(project);
    step.generateDefinitionContent(project);
    step.generateRequestContent(spec, project);
    step.writeProject(project, tsGearConfigPath);
    projectGlobalVariable_1.restore(project);
};
/**
 * run step by step
 * sequence could not be changed
 * every step depends on the pre step
 * */
exports.run = async () => {
    const { projects, tsGearConfigPath } = await step.getUserConfig();
    projects.forEach((project) => exports.processProject(project, tsGearConfigPath));
};
