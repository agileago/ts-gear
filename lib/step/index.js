"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./getUserConfig"));
__export(require("./fetchSwagger"));
__export(require("./prepareProjectDirectory"));
__export(require("./translateSchema"));
__export(require("./assembleSchemaToGlobal"));
__export(require("./cleanRefAndDefinitionName"));
__export(require("./parseGenericType"));
__export(require("./colletRefsInRequestAndPatchDefinition"));
__export(require("./generateDefinitionContent"));
__export(require("./generateRequestContent"));
__export(require("./writeProject"));
