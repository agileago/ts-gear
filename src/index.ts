export { requester as fetchRequester } from './requester/fetch'
export { requester as axiosRequester } from './requester/axios'
export { IProject, PropertyType } from './interface'
