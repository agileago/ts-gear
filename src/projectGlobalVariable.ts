import type { ProjectGlobalMap, Project } from './type'
import { clearObject } from './tool/clearObject'

const projectGlobal: ProjectGlobalMap = {}

// let currentProject: Project | undefined

export const getGlobal = (project: Project) => {
  if (!projectGlobal[project.name]) {
    projectGlobal[project.name] = {
      definitionMap: {},
      requestMap: {},
      requestRefSet: new Set(),
      requestEnumSet: new Set(),
      enumMap: {},
    }
  }
  return projectGlobal[project.name]
}

export const restore = (project: Project) => {
  const g = projectGlobal[project.name]
  clearObject(g.definitionMap)
  clearObject(g.requestMap)
  clearObject(g.enumMap)
  g.requestRefSet.clear()
  g.requestEnumSet.clear()
}
