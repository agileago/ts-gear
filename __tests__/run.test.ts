import { join } from 'path'

import { Spec } from 'swagger-schema-official'

import * as step from 'src/step'
import { Project } from 'src/type'
import { getGlobal } from 'src/projectGlobalVariable'
import { run } from 'src/run'

describe('run', () => {
  it('with fixture', async () => {
    const cwd = process.cwd()
    // change cwd to `example`
    process.chdir(join(cwd, 'example', 'petProject'))
    try {
      await run()
    } catch (e) {
      console.log(e)
    }
    process.chdir(cwd)
  })

  it('use cleaned spec', async () => {
    const spec = {
      swagger: '2.0',
      info: {
        description: '',
        version: '1.0',
        title: '接口文档',
        termsOfService: 'http://localhost',
        contact: { name: '管理平台', url: 'http://localhost', email: '67890211@qq.com' },
        license: { name: 'license', url: 'http://localhost' },
      },
      host: 'http://localhost',
      basePath: '/',
      tags: [{ name: 'credit-query-controller', description: 'Credit Query Controller' }],
      paths: {
        '/api/core/asset/credit/query/pastCreditCardBillGather': {
          post: {
            tags: ['credit-query-controller'],
            summary: '往期账单汇总',
            description: '往期账单汇总',
            operationId: 'pastCreditCardBillGatherUsingPOST',
            consumes: ['application/json;charset=UTF-8'],
            produces: ['application/json;charset=UTF-8'],
            parameters: [
              {
                in: 'body',
                name: 'body',
                description: 'requestParamVo',
                required: true,
                schema: { $ref: 'GeneralRequestParameterQueryParameter' },
              },
            ],
            responses: {
              '200': { description: 'OK', schema: { $ref: 'DataTransOutput' } },
              '201': { description: 'Created' },
              '401': { description: 'Unauthorized' },
              '403': { description: 'Forbidden' },
              '404': { description: 'Not Found' },
            },
          },
        },
        '/zzjy/all-msg/getAllMsgForMap': {
          post: {
            tags: ['credit-query-controller'],
            summary: '站址经营-汇总首页',
            operationId: 'getAllMsgForMapUsingPOST',
            consumes: ['application/json'],
            produces: ['*/*'],
            parameters: [
              { name: 'accountTime', in: 'query', description: 'accountTime', required: true, type: 'string' },
              { name: 'type', in: 'query', description: 'type', required: true, type: 'string' },
            ],
            responses: {
              '200': { description: 'OK', schema: { $ref: 'ResultListMap' } },
              '201': { description: 'Created' },
              '401': { description: 'Unauthorized' },
              '403': { description: 'Forbidden' },
              '404': { description: 'Not Found' },
            },
          },
        },
      },
      definitions: {
        DataTransOutput: {
          type: 'object',
          required: ['transCode', 'transMessage', 'transMessageDetail'],
          properties: {
            data: { type: 'object', description: '返回数据' },
            transCode: {
              type: 'integer',
              format: 'int32',
              description: '错误码。\n100000 成功\n200000 入参不合法\n400000 权限不足\n500000 服务失败',
            },
            transMessage: { type: 'string', description: '错误信息。成功：“成功” 失败：“失败对应的msg”' },
            transMessageDetail: { type: 'string', description: '信息详情”' },
          },
          description: '带数据的返回数据',
        },
        ResultListMap: {
          type: 'object',
          properties: {
            description: { type: 'string' },
            result: { type: 'array', items: { $ref: 'Map' } },
            returnCode: { type: 'integer', format: 'int32' },
          },
        },
        OutputParameterVo: {
          type: 'object',
          required: ['dueDay', 'requestChannel'],
          properties: {
            dueDay: { type: 'string', description: '开卡日' },
            requestChannel: { type: 'string', description: '进件渠道' },
          },
        },
        ABCOutputParameters: {
          type: 'object',
          required: ['dueDay', 'requestChannel'],
          properties: {
            dueDay: { type: 'string', description: '开卡日' },
            requestChannel: { type: 'string', description: '进件渠道' },
          },
        },
        MixedChineseAndEnglishWithSpacesVo: {
          type: 'object',
          required: ['dueDay', 'requestChannel'],
          properties: {
            dueDay: { type: 'string', description: '开卡日' },
            requestChannel: { type: 'string', description: '进件渠道' },
          },
        },
        QueryParameters: {
          type: 'object',
          required: ['dueDay', 'requestChannel'],
          properties: {
            dueDay: { type: 'string', description: '开卡日' },
            requestChannel: { type: 'string', description: '进件渠道' },
            data: { type: 'object', additionalProperties: { type: 'array', $ref: 'DataTransOutput' } },
          },
        },
        GeneralRequestParameterTokenOutputParameterVO: {
          type: 'object',
          properties: { bizParamVo: { $ref: 'OutputParameterVo' } },
        },
        DatatransoutputOutputParameterVO: {
          type: 'object',
          required: ['transCode', 'transMessage', 'transMessageDetail'],
          properties: {
            data: { description: '返回数据', $ref: 'OutputParameterVo' },
            transCode: {
              type: 'integer',
              format: 'int32',
              description: '错误码。\n100000 成功\n200000 入参不合法\n400000 权限不足\n500000 服务失败',
            },
            transMessage: { type: 'string', description: '错误信息。成功：“成功” 失败：“失败对应的msg”' },
            transMessageDetail: { type: 'string', description: '信息详情”' },
          },
          description: '带数据的返回数据',
        },
      },
    } as Spec
    const project: Project = {
      name: 'projectPont',
      source: 'fixture/pontFixture.json',
      dest: './service',
      // keepGeneric: true,
      translationEngine: 'baidu',
      requester: () => Promise.resolve({}),
    }
    step.assembleSchemaToGlobal(spec, project)
    step.parseGenericType(project)

    const { definitionMap } = getGlobal(project)
    expect(definitionMap).toMatchSnapshot()
  })
})
