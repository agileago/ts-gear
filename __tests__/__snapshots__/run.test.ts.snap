// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run use cleaned spec 1`] = `
Object {
  "ABCOutputParameters": Object {
    "schema": Object {
      "properties": Object {
        "dueDay": Object {
          "description": "开卡日",
          "type": "string",
        },
        "requestChannel": Object {
          "description": "进件渠道",
          "type": "string",
        },
      },
      "required": Array [
        "dueDay",
        "requestChannel",
      ],
      "type": "object",
    },
    "typeName": "ABCOutputParameters",
  },
  "DataTransOutput": Object {
    "schema": Object {
      "description": "带数据的返回数据",
      "properties": Object {
        "data": Object {
          "description": "返回数据",
          "type": "object",
        },
        "transCode": Object {
          "description": "错误码。
100000 成功
200000 入参不合法
400000 权限不足
500000 服务失败",
          "format": "int32",
          "type": "integer",
        },
        "transMessage": Object {
          "description": "错误信息。成功：“成功” 失败：“失败对应的msg”",
          "type": "string",
        },
        "transMessageDetail": Object {
          "description": "信息详情”",
          "type": "string",
        },
      },
      "required": Array [
        "transCode",
        "transMessage",
        "transMessageDetail",
      ],
      "type": "object",
    },
    "typeName": "DataTransOutput",
  },
  "DatatransoutputOutputParameterVO": Object {
    "schema": Object {
      "description": "带数据的返回数据",
      "properties": Object {
        "data": Object {
          "$ref": "OutputParameterVo",
          "description": "返回数据",
        },
        "transCode": Object {
          "description": "错误码。
100000 成功
200000 入参不合法
400000 权限不足
500000 服务失败",
          "format": "int32",
          "type": "integer",
        },
        "transMessage": Object {
          "description": "错误信息。成功：“成功” 失败：“失败对应的msg”",
          "type": "string",
        },
        "transMessageDetail": Object {
          "description": "信息详情”",
          "type": "string",
        },
      },
      "required": Array [
        "transCode",
        "transMessage",
        "transMessageDetail",
      ],
      "type": "object",
    },
    "typeName": "DatatransoutputOutputParameterVO",
  },
  "GeneralRequestParameterQueryParameter": Object {
    "typeName": "GeneralRequestParameterQueryParameter",
    "typescriptContent": "export type GeneralRequestParameterQueryParameter = any",
  },
  "GeneralRequestParameterTokenOutputParameterVO": Object {
    "schema": Object {
      "properties": Object {
        "bizParamVo": Object {
          "$ref": "OutputParameterVo",
        },
      },
      "type": "object",
    },
    "typeName": "GeneralRequestParameterTokenOutputParameterVO",
  },
  "MixedChineseAndEnglishWithSpacesVo": Object {
    "schema": Object {
      "properties": Object {
        "dueDay": Object {
          "description": "开卡日",
          "type": "string",
        },
        "requestChannel": Object {
          "description": "进件渠道",
          "type": "string",
        },
      },
      "required": Array [
        "dueDay",
        "requestChannel",
      ],
      "type": "object",
    },
    "typeName": "MixedChineseAndEnglishWithSpacesVo",
  },
  "OutputParameterVo": Object {
    "schema": Object {
      "properties": Object {
        "dueDay": Object {
          "description": "开卡日",
          "type": "string",
        },
        "requestChannel": Object {
          "description": "进件渠道",
          "type": "string",
        },
      },
      "required": Array [
        "dueDay",
        "requestChannel",
      ],
      "type": "object",
    },
    "typeName": "OutputParameterVo",
  },
  "QueryParameters": Object {
    "schema": Object {
      "properties": Object {
        "data": Object {
          "additionalProperties": Object {
            "$ref": "DataTransOutput",
            "type": "array",
          },
          "type": "object",
        },
        "dueDay": Object {
          "description": "开卡日",
          "type": "string",
        },
        "requestChannel": Object {
          "description": "进件渠道",
          "type": "string",
        },
      },
      "required": Array [
        "dueDay",
        "requestChannel",
      ],
      "type": "object",
    },
    "typeName": "QueryParameters",
  },
  "ResultListMap": Object {
    "schema": Object {
      "properties": Object {
        "description": Object {
          "type": "string",
        },
        "result": Object {
          "items": Object {
            "$ref": "Map",
          },
          "type": "array",
        },
        "returnCode": Object {
          "format": "int32",
          "type": "integer",
        },
      },
      "type": "object",
    },
    "typeName": "ResultListMap",
  },
}
`;
